includes/restoreMappings.js

```diff
@@ -0,0 +1,112 @@
+// Copyright © 2017, 2023 IBM Corp. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+'use strict';
+
+const error = require('./error.js');
+const debug = require('debug');
+
+const mappingDebug = debug('couchbackup:mappings');
+
+class Restore {
+  constructor(db) {
+    this.db = db;
+    this.totalDocsRestored = 0;
+    this.batchCounter = 0;
+  }
+
+  /**
+   * Mapper for converting a backup file line to an array of documents pending restoration.
+   *
+   * @param {object} backupLine object representation of a backup file line {lineNumber: #, line: '...'}
+   * @returns {array} array of documents parsed from the line
+   */
+  backupLineToDocsArray = async(backupLine) => {
+    if (backupLine && backupLine.line !== '') {
+      // see if it parses as JSON
+      try {
+        const arr = JSON.parse(backupLine.line);
+        // if it's an array with a length
+        if (Array.isArray(arr) && arr.length > 0) {
+          return arr;
+        } else {
+          return Promise.reject(new error.BackupError('BackupFileJsonError', `Error on line ${backupLine.lineNumber} of backup file - not an array`));
```

Comment:
It's a copy of what was there before, but yeah the error should probably only be for non-array content.
I'm not sure if it's actually possible to write an empty array to the backup file, but I think it could happen if there was a purge inbetween spooling changes and fetching the content. In any case we now use an empty array for broken lines etc when there is nothing to do so an actual empty array should just pass through (to be filtered out again later along with the blank line ones).

In line with your other comments I noticed that there was no need for this to be async either.

Fixed in 3f1bd35ea14dbf6572c493b9bba8668a442ae55c